jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/DefaultToolkitConnectionManager.kt

```diff
@@ -108,7 +108,7 @@ class DefaultToolkitConnectionManager : ToolkitConnectionManager, PersistentStat
             if (oldConnection != null && newConnection != null && pinningManager != null) {
                 val featuresToPin = mutableListOf<FeatureWithPinnedConnection>()
                 FeatureWithPinnedConnection.EP_NAME.forEachExtensionSafe {
-                    if (!pinningManager.isFeaturePinned(it) && it.supportsConnectionType(oldConnection) && !it.supportsConnectionType(newConnection)) {
+                    if (!pinningManager.isFeaturePinned(it) && (it.supportsConnectionType(oldConnection) != it.supportsConnectionType(newConnection))) {
```

Comment:
`it.supportsConnectionType(oldConnection) ` is this a boolean? Could you explain this change? Thanks!

Reply:
Or add a one sentence comment in this function

Reply:
yes it's a boolean, right now we only shows this prompt when users switching from BID/SSO to IAM, now we want to show it for switching from IAM to BID/SSO as well